explainer.md

```diff
@@ -68,356 +107,405 @@ If the set of pages being badged corresponds to an [Installed Web Application](h
-* Due to the nature of being a 16x16-pixel icon, the user agent must render the
-  text or number into an image. It pretty much has to be 1–2 characters.
-* Flag badges are just passed as a coloured circle.
-* Integer badges are rendered as the number, if a single digit, or "+", if
-  greater than 9.
+### The model
 
-Note: This API only allows badging a currently open window.
+A badge is associated with a [scope](https://www.w3.org/TR/appmanifest/#navigation-scope).
 
-### Android
+* Documents are badged with the most specific badge for their URL (i.e. prefer a badge for `/page/1` to a badge for `/page/` when on the url `/page/1?foo=7`).
+* For [installed applications](https://www.w3.org/TR/appmanifest/#installable-web-applications), a user agent **MAY** display the badge with the most specific scope still encompassing the [navigation scope](https://www.w3.org/TR/appmanifest/#navigation-scope) of the application in an [OS specific context](#OS-Specific-Contexts).
 
-Requires Android 8.0 (Oreo) or higher.
+At any time, the badge for a specific scope, if it is set, may be either:
 
-* The host API is
-  [`NotificationChannel`](https://developer.android.com/training/notify-user/badges),
-  which lets you set a number only. The badge is usually shown as a coloured
-  dot; the number is only shown on a long-press.
-* Tricky: The API is tied to notifications. You can't show a badge unless there
-  are pending notifications.
+* A "flag" indicating the presence of a badge with no contents, or
+* A positive integer.
 
-### Chrome OS
+The model does not allow a badge to be a negative integer, or the integer value 0 (setting the badge to 0 is equivalent to clearing the badge).
 
-Not currently supported, but will [soon be available](https://crbug.com/801014)
-for Android apps using the above `NotificationChannel` API. The Chrome browser
-itself will be able to re-use this mechanism for showing a coloured dot on web
-application icons.
+### The API
 
-### iOS
+The `Badge` interface is a member object on
+[`Window`](https://html.spec.whatwg.org/#the-window-object) and
+[`ServiceWorkerGlobalScope`](https://w3c.github.io/ServiceWorker/#serviceworkerglobalscope-interface).
+It contains the following methods:
 
-* The host API is
-  [`UIApplication.applicationIconBadgeNumber`](https://developer.apple.com/documentation/uikit/uiapplication/1622918-applicationiconbadgenumber),
-  which lets you set a positive integer only.
-* Integer badges are passed directly to the host API.
-* Flag badges and strings other than digits are just represented by the number
-  "1".
+* `Badge.set([contents], [options])`: Sets the badge for the scope in *options*
+  to *contents* (an integer), or to "flag" if *contents* is omitted. If
+  *contents* is 0, clears the badge for the given scope.
+* `Badge.clear([options])`: Clears the badge for the scope in *options*.
+* `Badge.canBadgeDocument()`: Returns true if the user agent would display a
```

Comment:
```suggestion
* `Badge.canBadgeDocument()`: Returns true if the user agent can display a
    document scoped badge on or near the page's favicon.
```

Reply:
I think would is better here. If it merely can, then you have no way of knowing whether it did.

I guess your API is better where it just returns what it *did* do. But since this is a strawman proposal, I'll leave canBadgeDocument for now since I don't want to design a second strawman proposal for feature detection.